### ISSUE TITLE
MW4 used with GM4000

### ISSUE THREAD

**Issue by mworion** [mworion/MountWizzard4#88]
Dear Michael,
with much delight I have discovered your software for modelling 10micron mounts.
We would like to use it to model our observatories GM4000HPS for imaging with INDI/Ekos/kstars.

Let me briefly describe the setup we use.
- Mount: 10micron GM4000HPS, firmware version 2.16.12, ethernet connection, IP 169.254.82.111
- Scope: Planewave CDK20, FL 3454mm, f/6.9
- Main Camera: SBIG STX-16803
- Computer/Software: Linux, Ubuntu 20.04, XFCE for local login, LXDE for VNC (remotely from the offices in the next building, it's warmer there), imaging is done with INDI/Ekos/kstars

I would like to ask a few questions, if this is fine with you, and report a potential bug.
We have to provide the astronomy lab course over the winter now remotely (I think it's obvious why), and to have as few problems as possible we yesterday wanted to create a new model for the mount with at least 70 points using MountWizzard4.
The installation worked very nicely, the connection to the mount, openweathermap and the camera provided by the indi server launched by ekos was also fine.
For plate solving we use a local version of astrometry.net with all index files installed, which was also found by MountWizzard4.

Now, I have a question about the plate solving process using MountWizzard4 in combination with ekos. Is it correct that in order for MountWizzard4 to properly solve an image the WCS coordinates in the image files provided by the indi server must contain the WCS keywords for the initial guess of the coordinates? When we do not connect the mount also to ekos and solve at least one image there such that it writes the WCS keywords to the fits file, MountWizzard4 assumes all coordinates are zero, thus also the plate solving fails. Is it okay to connect the mount to two clients (ekos/indi and MountWizzard4) at once?

In order to model the mount we first cleared the old model, created a horizon mask, generated 70-some-odd model points, and started the modelling process. MountWizzard4 slewed the mount to the first point, took an image with the indi server, and the queued it for plate solving which also worked. However, then something strange happened. In the message window it said
\t22:17:26 Modeling start:\tm-2020-11-27-21-17-21-align3
\t22:17:26 Slewing mount:\t\tpoint 001, altitude\t55, azimuth:\t10
\t22:17:26 Align model data refreshed
\t22:18:02 Exposing image-001:\tpath: image-001.fits
\t22:18:05 Slewing mount:\t\tpoint 002, altitude\t55, azimuth:\t130
\t22:18:06 Slewing mount:\t\tpoint 003, altitude\t55, azimuth:\t250
\t22:18:14 Solving image-001:\tpath: image-001.fits
\t22:19:00 Exposing image-002:\tpath: image-002.fits
so as you can see it does the first point correctly, then says it slews to point 2 but then immediately, only a second later, slews to point 3. Having arrived at point 3, it takes an image, but treats this image as if it would origin from point 2. When this image is solved it also marks point 2 on the hemisphere plot as done. The same now happens for point 4 and point 5, so it says, it slews to point 4, but then immediately slews to point 5, takes an image, and then treats this as the image from point 3 (yes, 3, not 4).
I think it is obvious that this is not what is intended.
Below are the corresponding lines from the deep debug log, the complete log can be found at: https://weber.whyisitso.de/owncloud/index.php/s/6CuyaPUFuH9hf8M
I would kindly like to ask you whether we just did something wrong or whether there is maybe something wrong with the software itself.
Kind regards and thank you in advance,
Philipp Weber

======================
Log:
[2020-11-27 21:17:24.677][I][ mainApp.py][ 458] Message window: [Modeling start: m-2020-11-27-21-17-21-align3]
[2020-11-27 21:17:24.678][I][ tabModel.py][ 378] Slew started
[2020-11-27 21:17:25.162][I][ tabModel.py][ 420] Queued to image [001]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-001.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 1, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 10}]
[2020-11-27 21:17:25.162][I][ mainApp.py][ 458] Message window: [Slewing mount: point: 001, altitude: 55, azimuth: 10]
[2020-11-27 21:17:25.521][I][ mainApp.py][ 458] Message window: [Align model data refreshed]
[2020-11-27 21:17:27.700][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:17:38.701][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:17:49.802][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:17:57.615][W][ loader.py][ 192] Click Object : []
[2020-11-27 21:17:57.616][W][ loader.py][ 192] Click Object : [ImageDialogWindow]
[2020-11-27 21:17:57.650][W][ loader.py][ 192] Click Object : []
[2020-11-27 21:17:57.651][W][ loader.py][ 192] Click Object : [ImageDialogWindow]
[2020-11-27 21:18:01.011][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:18:01.771][I][ tabModel.py][ 327] Imaging started
[2020-11-27 21:18:01.772][I][ cameraIndi.py][ 220] camera has readout quality entry: {}
[2020-11-27 21:18:01.772][I][ cameraIndi.py][ 269] Camera has no download quality settings
[2020-11-27 21:18:01.773][I][ tabModel.py][ 356] Queued to solve [001]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-001.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 1, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 10, 'raJNowM': <Angle 05h 33m 16.27s>, 'decJNowM': <Angle 82deg 01' 34.0">, 'angularPosRA': <Angle 134deg 18' 38.9">, 'angularPosDEC': <Angle 07deg 58' 50.2">, 'siderealTime': <Angle 02h 30m 15.01s>, 'julianDate': <Time tt=2459181.388285933>, 'pierside': 'W'}]
[2020-11-27 21:18:01.773][I][ mainApp.py][ 458] Message window: [Exposing image-001: path: image-001.fits]
[2020-11-27 21:18:04.778][I][ tabModel.py][ 378] Slew started
[2020-11-27 21:18:05.095][I][ tabModel.py][ 420] Queued to image [002]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-002.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 2, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 130}]
[2020-11-27 21:18:05.095][I][ mainApp.py][ 458] Message window: [Slewing mount: point: 002, altitude: 55, azimuth: 130]
[2020-11-27 21:18:05.778][I][ tabModel.py][ 378] Slew started
[2020-11-27 21:18:06.020][I][ tabModel.py][ 420] Queued to image [003]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-003.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 3, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 250}]
[2020-11-27 21:18:06.020][I][ mainApp.py][ 458] Message window: [Slewing mount: point: 003, altitude: 55, azimuth: 250]
[2020-11-27 21:18:07.035][I][ tabModel.py][ 378] Slew started
[2020-11-27 21:18:07.035][W][ tabModel.py][ 381] Empty slew queue
[2020-11-27 21:18:12.844][W][ cameraIndi.py][ 200] Image BLOB is uncompressed fits format
[2020-11-27 21:18:12.844][I][ tabModel.py][ 284] Solving started
[2020-11-27 21:18:13.434][I][ tabModel.py][ 295] Queued to result [001]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-001.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 1, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 10, 'raJNowM': <Angle 05h 33m 16.27s>, 'decJNowM': <Angle 82deg 01' 34.0">, 'angularPosRA': <Angle 134deg 18' 38.9">, 'angularPosDEC': <Angle 07deg 58' 50.2">, 'siderealTime': <Angle 02h 30m 15.01s>, 'julianDate': <Time tt=2459181.388285933>, 'pierside': 'W'}]
[2020-11-27 21:18:13.440][I][ mainApp.py][ 458] Message window: [Solving image-001: path: image-001.fits]
[2020-11-27 21:18:13.800][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:18:15.683][I][astrometryNET.p][ 137] IMAGE2XY took 2.2378485202789307s return code: 0 [/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-001.fits] stderr: stdout: simplexy: found 802 sources. 
[2020-11-27 21:18:15.685][I][ astrometry.py][ 111] Header RA: 05h 28m 49.41s (82.20589), DEC: 82deg 00' 45.6" (82.01266), Scale: 1.075101
[2020-11-27 21:18:26.105][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:18:37.773][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:18:41.325][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:41.325][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:41.762][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:41.762][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:41.936][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:41.936][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:42.496][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:42.496][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:42.536][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:42.537][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:42.562][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:42.562][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:43.154][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:43.154][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:43.224][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:43.224][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:43.844][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:43.844][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:43.970][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:43.970][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.080][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.080][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.180][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.180][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.376][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.376][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.452][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.452][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.528][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.528][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.596][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.596][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.628][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.628][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.983][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.983][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:44.988][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:45.032][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:45.032][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:45.096][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:45.096][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:45.128][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:45.128][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:45.196][W][ loader.py][ 192] Click Object : [qt_scrollarea_viewport]
[2020-11-27 21:18:45.196][W][ loader.py][ 192] Click Object : [MessageDialogWindow]
[2020-11-27 21:18:50.685][I][onlineWeather.p][ 282] http://api.openweathermap.org/data/2.5/forecast?lat=49.83&lon=10.83&APPID=693a39dccf3512f938b4898322e10317
[2020-11-27 21:18:59.134][I][ tabModel.py][ 327] Imaging started
[2020-11-27 21:18:59.134][I][ cameraIndi.py][ 220] camera has readout quality entry: {}
[2020-11-27 21:18:59.134][I][ cameraIndi.py][ 269] Camera has no download quality settings
[2020-11-27 21:18:59.136][I][ tabModel.py][ 356] Queued to solve [002]: [{'imagePath': '/home/observer/Desktop/MW4/image/m-2020-11-27-21-17-21-align3/image-002.fits', 'exposureTime': 3.0, 'binning': 2.0, 'subFrame': 100.0, 'fastReadout': False, 'lenSequence': 3, 'countSequence': 2, 'modelName': 'm-2020-11-27-21-17-21-align3', 'astrometryApp': 'astrometry - ASTROMETRY.NET', 'solveTimeout': 60.0, 'searchRadius': 10.0, 'focalLength': 3454.0, 'altitude': 55, 'azimuth': 130, 'raJNowM': <Angle 23h 56m 24.14s>, 'decJNowM': <Angle 29deg 57' 15.1">, 'angularPosRA': <Angle 38deg 41' 24.7">, 'angularPosDEC': <Angle -60deg 02' 15.4">, 'siderealTime': <Angle 02h 31m 12.18s>, 'julianDate': <Time tt=2459181.388944693>, 'pierside': 'E'}]
[2020-11-27 21:18:59.136][I][ mainApp.py][ 458] Message window: [Exposing image-002: path: image-002.fits]

**Comment by mworion** [comment]
Hi Philipp,
to you questions:

Is it correct that in order for MountWizzard4 to properly solve an image the WCS coordinates in the image files provided by the indi server must contain the WCS keywords for the initial guess of the coordinates? 

- No. MW does no need WCS coordinates as these drop off the plate solving process through local solver in your case. MW4 (or better the plate solver chosen) only needs a rough guess of the pointing coordinates of the telescope.

When we do not connect the mount also to ekos and solve at least one image there such that it writes the WCS keywords to the fits file, MountWizzard4 assumes all coordinates are zero, thus also the plate solving fails. Is it okay to connect the mount to two clients (ekos/indi and MountWizzard4) at once?

- Yes you could connect multiple clients to the server at the save time. Imaging is done though the INDI server and the Imaging part on server also looks for the coordinates of the mount. Therefore also the mount has to be connected on Indi server. 
- If you see no coordinates e.g. in the image window of MW4, then most probably the telescope driver is not connected and setup on Indi server. Right now MW4 wold be able to look up the coordinates itself, but it uses the coordinates provided in the fits header from Indi server.
- For explanation: the telescope driver in MW4 has only one reason: to read out the aperture and focallenght as MW4 directly talks to the mount without a driver.

The observation of the split I will check in the full log. But most probably you observed the async behavior of MW4. The image number, slew number, solved number need not to be in order as they are queue and processed separately .

Michel

**Comment by comptonizing** [comment]
Hi Michael,
as promised, attached you can find the entire log.
[mw4-2020-11-27.log](https://github.com/mworion/MountWizzard4/files/5610915/mw4-2020-11-27.log)
Philipp

**Comment by comptonizing** [comment]
Hi,
thank you for taking the time to answer my questions.

>  * If you see no coordinates e.g. in the image window of MW4, then most probably the telescope driver is not connected and setup on Indi server. Right now MW4 wold be able to look up the coordinates itself, but it uses the coordinates provided in the fits header from Indi server.

I think what is confusing here is that you have to solve at least one image in ekos such that it writes the proper coordinates into the fits header which can then be used by MW.

> The observation of the split I will check in the full log. But most probably you observed the async behavior of MW4. The image number, slew number, solved number need not to be in order as they are queue and processed separately .

I understand that, but the thing is that MW did not even slew to point 2 but went straight for point 3. Also it said it would capture image 2 when the scope was pointed at point 3, but afterwards made a tickmark at point 2 on the hemisphere plot.

Thanks again,
Philipp

**Comment by mworion** [comment]
Philipp,

Ok, understood. Strange. I try to reproduce it as I do not know why this "overtaking" happens.
Michel

**Comment by comptonizing** [comment]
Thank you very much for your effort.
If you would like to do spme sort of remote session you can also let me know.
Let me also say that the readout and download time of the SBIG STX-16803 is really long. An image without binning takes 15 seconds to get to disk on the computer after the exposure stopped, for 2x2 binned images this time is shorter but not by much. Just in case this might be part of the issue.
Philipp

**Comment by mworion** [comment]
Philipp,
question: did you have the retry option on
![Bildschirmfoto 2020-11-29 um 11 08 16](https://user-images.githubusercontent.com/6632980/100538971-3c38ac00-3233-11eb-9fbf-d3c2ffb65776.png)
while having your observations (meaning any setting >0)
Michel

PS: If you have long download time. Please check if you camera has a fast download mode. For solving around 1500 pixel should be enough. I use binning if I have more. Depending on the conditions I go up to 5 seconds exposure time to get more stars visible.

**Comment by comptonizing** [comment]
Hi,
> while having your observations (meaning any setting >0)
yes, indeed we did. It was set to three, because we weren't sure whether we could keep up with manually rotating the dome, although there is the really nice option to just pause the run.

> PS: If you have long download time. Please check if you camera has a fast download mode. For solving around 1500 pixel should be enough. I use binning if I have more. Depending on the conditions I go up to 5 seconds exposure time to get more stars visible.
We already set binning to 2x2 and 3 s exposures. I think this camera has something called 'burst mode', so I think we will try setting this and increase binning to 3x3 and the exposure length to 5s.
Thank you so much,
Philipp

**Comment by mworion** [comment]
Philipp,

OK, then I have a corner use case I have to dig in: Long exposure and download time, no coordinates meaning timed out solved and recap of the session. I probably start the session before I have the missed points (as they are still in processing) available.

The recap feature was introduce by user wish because they might some clouds while having a very limited field of view. so in this case you might miss many model points and you have to start over and over again. If you are in normal conditions (in my setup during 5 years now) I newer hand the hard need to do this.

Michel

Please report if you go to setting "0", but still experiencing the issue, because then I have to search elsewhere.

**Comment by comptonizing** [comment]
Hi,


> Please report if you go to setting "0", but still experiencing the issue, because then I have to search elsewhere.

I'll do that as soon as possible.
Thank you very much for your support.
Philipp

**Comment by comptonizing** [comment]
Hi Michel,
on Sunday there was some clear sky so we gave it another try, this time setting the retry number to 0. Unfortunately the behavior is still the same.
When listening carefully one can sometimes even hear the mount starting a slew but then, after a second or so, starting a slew in a different direction.
We also decided to see what happens when we just let the run continue. Here is a screenshot of what this looks like:
![shot](https://user-images.githubusercontent.com/67904665/101017771-d322a280-356a-11eb-89cb-267fd70703bb.png)
As you can see, MW thinks it has done half of the hemisphere when in reality it is already at the last point, thus it's just waiting forever.
However, when hitting the 'End' button under 'Model build', we were at least able to use the created model with 34 points. The pointing is pretty good with this model, but it's still not enough for unguided imaging.
Thanks,
Philipp

**Comment by mworion** [comment]
OK there is something more to correct. Could you please post the log as well ?

Michel

**Comment by comptonizing** [comment]
Hi,

> OK there is something more to correct. Could you please post the log as well ? Michel

Of course, the entire log is attached below.
Thank you for your help,
Philipp

[mw4-2020-11-29.log](https://github.com/mworion/MountWizzard4/files/5638546/mw4-2020-11-29.log)

**Comment by mworion** [comment]
Philipp,
I think I found the reason. Needs to be corrected. I will do this till Saturday. It's a race condition with timing issue.
Please check if in you setup exposure time + slewing time between two points is less than the download time of the camera. As you mentioned this could be the case. If so, I start slewing already before image download has ended.
Michel

**Comment by mworion** [comment]
Hi Philipp,

I made a beta for you. It's 0.220.4b1. If you check beta in updater section in MW4, you should see this version. 
Please report back if that works for you.

Michel

**Comment by comptonizing** [comment]
Hi,
I am very pleased to hear this, than you so much! However, the weather currently is a disaster.
As soon as there is the possiblity I will test it out and let you know.
Thank you very much,
Philipp

**Comment by mworion** [comment]
Never mind, actually it's 0.220.4b4 as I move forward. But anyhow, please look for the newest beta before starting.
Michel

**Comment by mworion** [comment]
Hi Philipp,

how things going. Anything new about this issue. As I move to a first release, I just would like to collect feedback from all not closed issues.

Michel

**Comment by comptonizing** [comment]
Hi,
over the christmas break, until 6th of January, the observatory was closed, so for some weeks now nothing has been done. But the weather was terrible anyway.
What I can tell you already is that before the shutdown under cloudy skies we did a quick "dry run" to see whether in principle everything is working and I can verify that MW did not skip the second point. But obviously plate solving didn't work because of the weather so I can not tell whether everything would have continued correctly from this point.
If you are willing to we can leave this issue open for now and I will try to catch a useful night and verify the entire functionality.
Thank you very much for your patience and help.
Ciao,
Philipp

**Comment by mworion** [comment]
Phillip, 
ok we leave it open. As soon as you test 0.240.0, please post the log file here. Remark: the default setting of logging as going to release is standard, so please go manual to debug_trace.
Michel

**Comment by comptonizing** [comment]
Hi Michel,
yesterday we finally had the time to do an extensive test. With the current release, being 1.0.2, the problem persists. However, I quickly restored version 0.220.4b6 from our backups and with this version it worked, we were able to create a 95 point model.
The only weird thing was that for the first few points in the south the mount takes mulitple minutes to finish one slew. Do you happen to know anything about this?
Other than that everything is working perfectly.
Ciao and thank you for your patience and support,
Philipp

**Comment by mworion** [comment]
Philipp,

did you have a test with v1.0.7 official ?

**Comment by comptonizing** [comment]
Hi Michel,
unfortunately no, the last time we used MW4 it was shortly before I wrote the comment and using the MW internal update option the newest available release at the time was 1.0.2.
From your question I realize that a test using version 1.0.7 would be good so I will do that as soon as the weather is okay.  Currently it looks horrible in northern Bavaria, but I will come back to you as soon as there is a chance.
Let me thank you again for your incredible support.
Philipp

**Comment by mworion** [comment]
to be solved in 1.1.0 if necessary

**Comment by comptonizing** [comment]
Hi Michel,
as promised I did a test with version 1.0.7 when the weather was okay. Unfortunately in this version the problem persists.
Attached you can find the log.
Let me quickly remind you that with version 0.220.2 it worked just fine.
Ciao and thanks,
Philipp
[mw4-2021-03-31.log](https://github.com/mworion/MountWizzard4/files/6249052/mw4-2021-03-31.log)

**Comment by mworion** [comment]
Hi Philipp,
sorry to hear about this situation. I'm currently revising all topics again and try to reproduce the behavior you see. So I reopen the ticket.

First I have to correct my first answer as it might be not clear enough. MW4 does not need the full WCS information from EKOS as this is added through astrometry.net solver solution. But it needs a solving hint from EKOS (means at least RA / DEC / SCALE) for being able to transfer them to astrometry.net. Otherwise astrometry.net will normally time out (which happens in some cases in the log).

Second I see in you recent log an issue when a solve failed and coming back to model making. This happened in an other case, too and is solved in the actual MW4 release v1.1.0. This might not interfere with the problem you see an was also in in 0.220.2 where you referenced MW4 is working.

The difference I see so far might be related to the long download times. In 0.220.x due to an error (yes error) I defined "image integrated" as downloaded and solved. This is incorrect, because it will force MW4 to wait for the next slew until the image is processed (downloaded and saved), which could be done in parallel. On the other side this increased the time before MW4 would like to expose the next image and might influence the timing behavior for your setup.

So far my insight, I'm still in the phase of investigating.
Michel

**Comment by mworion** [comment]
Hi Philipp,

I think I got it. At least it would explain so far the behavior you experienced:
My Camera, as well as the simulator sends this line only once:

`[2021-04-02 20:58:20.375][T][    indiBase.py][ 973] RecvCmd: [setNumberVector (CCD_EXPOSURE, CCD Simulator, Busy, 60)   oneNumber (CCD_EXPOSURE_VALUE) 0`

Your camera send this line two times. Due to this fact I send the "integration finished" signal twice (as camera is busy, but exposure = 0 is the only constraint to determine integration finished downloading started). Sending the signal moves the modeling procedure to slew o the next point. As stated in the last post, this was covered as I sent the signal in the past falsely at the finished download state, which "healed" this behavior.

The actual release v1.1.1 will includes this fix as is available Saturday. Hopefully this cleans the situation up.

Michel

**Comment by comptonizing** [comment]
Hi Michel,
thank you very much for your work once again!
I just saw that version 1.1.1 was released. As soon as there is a chance I will have a test again.
Do you think this is something we should mention to or ask the INDI team about? It sounds kind of weird.
Thank you so much for your support.
Ciao,
Philipp

**Comment by mworion** [comment]
Philipp,
I have to deal with the robustness. Is it good to send the same data twice without any change ? No. Is the data wrong ? No as well. So from your side you could address it, but no one will take care about refinement I think.
Michel

**Comment by comptonizing** [comment]
Hi Michel,
it has been a very long time and I'm so sorry for that. Due to a variety of reasons we had to wait until yesterday to perform another test. We used version 1.1.1 which is, as far as I can tell, the newest one. The problem we were observing up to this point with skipping a point appears to be gone, we are very grateful for that.
Plate Solving failed quite often but this might have been related to clouds passing through. However, something a bit odd was happening. Whenever an image failed to solve the next image immediately was also reported as failed without the solver even running. That didn't seem correct. I don't think I have logs with the highest log level available, only the normal one.
Thank you very much,
Philipp

**Comment by mworion** [comment]
Hi Philipp,
I’m currently on the way finalizing v2.0. This has a lot of improvements and testing. This relates also to the way of reiterating missing points etc. I would recommend to test v2.0 beta (ideally separate install in new work directory). What do you think ?
Michel

**Comment by comptonizing** [comment]
Hi Michel,
I'd be happy to do that! Indeed, up to this point I always installed MW4 from scratch in a new directory which was very easy when using the installation scripts. Can I use the same mechanism to install the beta version?
Philipp

**Comment by mworion** [comment]
Philipp,
yes, please download the newest scripts and install MW4. As the update will firstly install the regular release, you need to run UpdateBeta afterward to get the newest beta. From there on the internal updater should now be able to handle any further updates /downgrades within MW4 itself.
Actual beta is v2.0.0b26. Still not perfect, but in a good shape.
Michel

**Comment by comptonizing** [comment]
Hi Michel,
thank you very much, I just installed the beta version and if the weather forecast is correct we will be able to try it tomorrow. I'm very looking forward to it!
Thanks,
Philipp

**Comment by comptonizing** [comment]
Hi Michel,
I am incredibly happy to report that yesterday we ran a model with 95 points without any problems. I can not bring across how grateful we are!
We configured the dome which I wrote a custom indi driver for and MW4 just went ahead and did everything completely on its own and finished the modelling in about an hour. This is really amazing, thank you so much!
There are, however, some minor points which I would like to address:
- Our dome has an opening of roughly 1.6m but MW4 only let us configure 1.0m at most
- We couldn't enable the online weather for some reason even though the openweathermap API setting was green
- Solved points from a previous run are still in the hemisphere plot. This is not a problem it just confused me a bit
- Sometimes for a very short period of time the scope and slit drawing in the hemisphere window disappear
- If the solver is not set up in the devices menu the start model butten was grayed out. This makes total sense, but maybe some sort of message would be nice. It took me quite some time to figure out what I did wrong
- One thing we couldn't test because every single image was solved perfectly: In the last version when an image failed to solve the next one also failed immediately without it even trying

For some reason I can not attach files to this comment so I created a link where you can download the log file of yesterday's session if you want to: [https://www.sternwarte.uni-erlangen.de/~weber/MW/mw4-2021-07-20.log](https://www.sternwarte.uni-erlangen.de/~weber/MW/mw4-2021-07-20.log)

Once again I'd like to express my extreme gratitude for what you created here!
Ciao,
Philipp

**Comment by mworion** [comment]
Hi Philipp,

good to hear it work now. I'm going to close the issue. To you other points:

- Our dome has an opening of roughly 1.6m but MW4 only let us configure 1.0m at most

next beta and following will have an increased limit of 2m

- We couldn't enable the online weather for some reason even though the openweathermap API setting was green

Your actual problem is that the API key does not work - it's wrong. The GUI will be changed in next beta.

- Solved points from a previous run are still in the hemisphere plot. This is not a problem it just confused me a bit

Tricky as I need them for possible retries. 

- Sometimes for a very short period of time the scope and slit drawing in the hemisphere window disappear

Have to observe this. Might be a performance issue.

- If the solver is not set up in the devices menu the start model butten was grayed out. This makes total sense, but maybe some sort of message would be nice. It took me quite some time to figure out what I did wrong

Status grey is described in tooltips.

- One thing we couldn't test because every single image was solved perfectly: In the last version when an image failed to solve the next one also failed immediately without it even trying

I take this as it is  :-)

Michel
